File: aragog-common/src/main/java/com/gongshw/aragog/common/model/Rule.java
/*
 * Copyright (C) 2016 Baidu, Inc. All Rights Reserved.
 */
package com.gongshw.aragog.common.model;

import java.util.List;

import lombok.Data;

/**
 * @author gongshiwei
 */
@Data
public class Rule {
    private long id;
    private String uuid;
    private String name;
    private List<String> listPageUrlPatterns;
    private List<String> detailPageUrlPatterns;
    private ScriptType scriptType;
    private String environmentInitScript;
    private String listPageParserScript;
    private String detailPageParserScript;
}

File: aragog-common/src/main/kotlin/com/gongshw/aragog/common/model/Rule.kt
/*
 * Copyright (C) 2016 Baidu, Inc. All Rights Reserved.
 */
package com.gongshw.aragog.common.model;

import java.util.List;

/**
 * @author gongshiwei
 */
data class Rule(
        val id: Long,
        val uuid: String,
        val name: String,
        val listPageUrlPatterns: List<String>,
        val detailPageUrlPatterns: List<String>,
        val scriptType: ScriptType,
        val environmentInitScript: String,
        val listPageParserScript: String,
        val detailPageParserScript: String
)